apps/meteor/ee/app/livechat-enterprise/server/jobs/AbstractOmniSchedulerClass.ts

```diff
@@ -0,0 +1,44 @@
+import Agenda from 'agenda';
+import { MongoInternals } from 'meteor/mongo';
+import { Collection } from 'mongodb';
+
+import { schedulerLogger } from '../lib/logger';
+
+const SCHEDULER_NAME = 'omnichannel_scheduler';
+
+export abstract class AbstractOmniSchedulerClass {
+	scheduler: Agenda;
+
+	abstract createJobDefinition(): void;
+
+	abstract createIndexes(collection: Collection): void;
```

Comment:
What if we narrow down what this function can do? I think we can implement createIndexes right here and receive an array/object with the indexes to be created and options. That way, the function can only do what we want it to do :eyes: